go/bind/extension.go

```diff
@@ -0,0 +1,531 @@
+package keybase
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"runtime/debug"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/encrypteddb"
+	"github.com/keybase/client/go/kbconst"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+
+	"github.com/keybase/client/go/chat"
+	"github.com/keybase/client/go/chat/attachments"
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/storage"
+	"github.com/keybase/client/go/externals"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/service"
+	"github.com/keybase/client/go/uidmap"
+	context "golang.org/x/net/context"
+)
+
+var extensionRi chat1.RemoteClient
+var extensionInited bool
+var extensionInitMu sync.Mutex
+
+func ExtensionIsInited() bool {
+	extensionInitMu.Lock()
+	defer extensionInitMu.Unlock()
+	return extensionInited
+}
+
+func ExtensionInit(homeDir string, mobileSharedHome string, logFile string, runModeStr string,
+	accessGroupOverride bool, externalDNSNSFetcher ExternalDNSNSFetcher, nvh NativeVideoHelper) (err error) {
```

Comment:
why do we need `ExternalDNSNSFetcher` and `NativeVideoHelper` if we just pass in `NULL` from native?

Reply:
It's for Android, I can just call into native directly from Go on iOS.

Reply:
Although I guess for the extension there is no Android so I could get rid of this.